xds/internal/xdsclient/tests/lds_watchers_test.go

```diff
@@ -912,18 +936,156 @@ func (s) TestLDSWatch_NACKError(t *testing.T) {
 
 	// Verify that the expected error is propagated to the watcher.
 	u, err := lw.updateCh.Receive(ctx)
+	if err != nil {
+		t.Fatalf("Timeout when waiting for a listener resource from the management server: %v", err)
+	}
+	gotErr := u.(listenerUpdateErrTuple).err
+	if gotErr == nil || !strings.Contains(gotErr.Error(), wantListenerNACKErr) {
+		t.Fatalf("Update received with error: %v, want %q", gotErr, wantListenerNACKErr)
+	}
```

Comment:
Should we implement a helper for this?

```golang
func verifyListenerError(ctx context.Context, updateCh *testutils.Channel, wantErr string) error {
	u, err := updateCh.Receive(ctx)
	if err != nil {
		return fmt.Errorf("timeout when waiting for a listener update from the management server: %v", err)
	}
	gotErr := u.(listenerUpdateErrTuple).err
	if gotErr == nil || !strings.Contains(gotErr.Error(), wantErr) {
		return fmt.Errorf("update received with error: %v, want %q", gotErr, wantErr)
	}
}
```

Reply:
Done. Though we need the same helper for all other resource types as well. Will send a separate PR.